### PR TITLE
Add EOL to compat container logs

### PR DESCRIPTION
**Pull request by crunchtime-ali**
Per request of @rhatdan I created the change mentioned by @psakar in https://github.com/containers/podman/issues/8058 as a pull request.

It fixes the problem that container logs provided by REST API do not contain an EOL (\n)

### CODE REVIEW

**Comment by rhatdan** [comment]
Thanks @crunchtime-ali 
Could you sign your commits.
git commit -a -s --amend
git push --force

LGTM
Should we have similar in the libpod interface?

@jwhonce @baude @containers/podman-maintainers PTAL

**Comment by mheon** [comment]
@ashley-cui PTAL

**Review by saschagrunert — APPROVED** [review]
LGTM

File: pkg/api/handlers/compat/containers_logs.go
Comment (vrothberg) [inline pkg/api/handlers/compat/containers_logs.go:151]: I think we should only do it for the compat layer:
```golang
if !utils.IsLibpodRequest(r) {
....
```

@ashley-cui WDYT?

Comment (ashley-cui) [reply pkg/api/handlers/compat/containers_logs.go:151]: That sounds good to me

Comment (crunchtime-ali) [reply pkg/api/handlers/compat/containers_logs.go:151]: @vrothberg @ashley-cui do you rather want:

```go
		if utils.IsLibpodRequest(r) {
			frame.WriteString(line.Msg)
		} else {
			frame.WriteString(line.Msg + "\n")
		}
``` 

or 

```
		frame.WriteString(line.Msg)
		if !utils.IsLibpodRequest(r) {
			frame.WriteString("\n")
		}
```

Comment (vrothberg) [reply pkg/api/handlers/compat/containers_logs.go:151]: The latter seems cleaner to me. Also please add a code comment (ideally with a link to the issue you're fixing) as well.

Comment (crunchtime-ali) [reply pkg/api/handlers/compat/containers_logs.go]: Added and squashed @vrothberg

**Comment by rhatdan** [comment]
/lgtm
/approve
/hold
@crunchtime-ali Thanks.

**Comment by TomSweeneyRedHat** [comment]
looks like a timing issue is meeting up the works.  @edsantiago is this one of your known "friends"?

```
[+0008s] ok 52 [01-basic] GET info : status=200
[+0008s] not ok 53 [01-basic] Time for ten /info requests
[+0008s] #  expected: <= 7 seconds
[+0008s] #    actual: 8 seconds
```

**Comment by TomSweeneyRedHat** [comment]
LGTM
once test gremlins are chased away

**Comment by edsantiago** [comment]
`[+0008s] not ok 53 [01-basic] Time for ten /info requests` - addressed in current master. 

Two other failures are #8068 which @cevich claims "should retry automatically" so I'm not going to press Re-run

**Comment by jwhonce** [comment]
restarted tests, agent stopped responding

**Comment by cevich** [comment]
> should retry automatically

Cirrus-CI gives you one freebie restart if the agent konks out for some reason.   Anything beyond that requires pressing the button :disappointed:

**Pull request merged**
